src/schemaGenerator.ts

```diff
@@ -678,51 +685,39 @@ function attachDirectiveResolvers(
       `Expected directiveResolvers to be of type object, got ${typeof directiveResolvers}`,
     );
   }
+
   if (Array.isArray(directiveResolvers)) {
     throw new Error(
       'Expected directiveResolvers to be of type object, got Array',
     );
   }
-  forEachField(schema, (field: GraphQLField<any, any>) => {
-    const directives = field.astNode.directives;
-    directives.forEach((directive: DirectiveNode) => {
-      const directiveName = directive.name.value;
-      const resolver = directiveResolvers[directiveName];
 
-      if (resolver) {
-        const originalResolver = field.resolve || defaultFieldResolver;
-        const Directive = schema.getDirective(directiveName);
-        if (typeof Directive === 'undefined') {
-          throw new Error(
-            `Directive @${directiveName} is undefined. ` +
-              'Please define in schema before using',
-          );
-        }
-        const directiveArgs = getArgumentValues(Directive, directive);
+  const directiveVisitors = Object.create(null);
 
+  Object.keys(directiveResolvers).forEach(directiveName => {
+    directiveVisitors[directiveName] = class extends SchemaDirectiveVisitor {
```

Comment:
The `directiveResolvers` option supported by `makeExecutableSchema` works exactly as it did before, but is now implemented in terms of `SchemaDirectiveVisitor.visitSchemaDirectives`.